README.md
# sakai-info #

last updated: 2014-12-12  
author: David Adams (daveadams@gmail.com)  
github url: https://github.com/daveadams/sakai-info

The *sakai-info* gem consists of a suite of Ruby libraries which enable the
exploration of a Sakai database without the intermediation of a Java VM or
any official Sakai code, along with a command line tool, `sin`.

Because the primary goal of this tool is to assist in information gathering
and troubleshooting, no capability to change the database is included in the
tool or the libraries.

## Testing ##

Tests are defined in ./test using Test::Unit. The default `rake` action is to
run all tests.

## Building ##

Use `rake` to test and build the gem:

    $ rake gem:build

The resulting gem will be saved to the working directory as
`sakai-info-0.5.7.gem`.

Cleanup built gems using:

    $ rake clean

## Installing ##

Install the *sakai-info* gem locally with:

    $ rake gem:install

Uninstall with:

    $ rake gem:uninstall

## Database Connectivity ##

[Sequel](http://sequel.rubyforge.org) is used for database connectivity. Driver
gems must also be installed to support whatever database or databases you use.

Oracle support requires the `ruby-oci8` gem, and MySQL support requires the
`mysql` gem. Some unit tests make use of the `sqlite3` gem.

## MySQL Support ##

MySQL support is now working. Your MySQL server must have the setting
`lower_case_table_names=1` or be running on a case-insensitive filesystem.

I recommend you use the `mysql2` gem, which is faster than the `mysql` or
`ruby-mysql` gems and is also much better tested.

## Configuration ##

To run, *sakai-info* needs to be able to connect to your Sakai database server.
It is possible to specify multiple instances to choose from at runtime.

In this release, *sakai-info* expects a to find the config in a file located at
`$HOME/.sakai-info`. The file must be in YAML format and can contain one or
more Sakai database connection nicknames and connection strings, for example:

    prod: oracle://sakai:password@SAKAIPROD
    test: mysql2://test:password@mysql-host/db_name
    local: mysql2://sakai:ironchef@localhost/db_name

For connections requiring options not well-supported by URI strings,
i.e. passwords with special characters, or non-default schemas, 
all options can be specified one their own line, eg:

    lmssbx:
      adapter: oracle
      user: user
      password: pas^w#?d
      host: db_name
      after_connect: ALTER SESSION SET CURRENT_SCHEMA=sakai

Some connections may require a query be run immediately after connecting, i.e.
to select a default schema. Sequel supports this through the after_connect
option. To provide a query for after_connect, include it as shown above.

The first connection in the list is the default connection. Other connections
may be specified using the corresponding YAML key, which functions as a
nickname for the connection.

[More information on how to specify a Sequel connection string.](http://sequel.rubyforge.org/rdoc/files/doc/opening_databases_rdoc.html)

As an alternative, may specify an alternate file name by putting an `@` sign
followed by the filename to load the config from on the first line of
the config file specified, eg:

    @/etc/sakai-info/shared.conf

## Command Line Usage ##

After installing the gem, the `sin` program should be found in your PATH. For
usage details, run:

    $ sin help

## Library Usage ##

To use the library in your own Ruby programs, simply specify:

    require 'sakai-info'

Further specifying `include SakaiInfo` will pull all object classes into the
primary namespace and could save some typing. Be careful that class names such
as `User`, `Site`, and `Group` do not conflict with other elements of your
program.

Full RDoc documentation for each class is not available in this release, but is
planned for a future release.

## Change History ##

See CHANGELOG.md

## Future Plans ##

See ROADMAP.md

## License ##

This work is dedicated to the public domain. No rights are reserved. See
LICENSE for more information.

## Contributors ##

The following individuals have contributed code to this project:

* David Adams <daveadams@gmail.com>
* Will Humphries <will.d.humphries@gmail.com>


CHANGELOG.md
# sakai-info Change History #

### 0.5.7 ###

*Released 2014-12-12*

* bugfix for "is auto submitted" field on QuizAttempt

### 0.5.6 ###

*Released 2014-12-11*

* Added submitted date and elapsed time to quiz attempt serialization
* Added additional shortcut names for assignment and assignment-submission
* New object types: pending-quiz-metadata, published-quiz-metadata
* New object types: pending-quiz-authorization, published-quiz-authorization

### 0.5.5 ###

*Released 2014-07-08*

* New object types: calendar and calendar-event
* Added Site.find_all
* Added mnemonic descriptions to quiz attempt status field

### 0.5.4 ###

*Released 2013-06-28*

* More Sequel 4.0 fixes for deprecated sql_function and Symbol#like

### 0.5.3 ###

*Released 2013-06-28*

* Unit test cleanup, refactor, and migration to MiniTest
* Added simple tests against user/site fixture data
* Refactored and simplified rake tasks for initializing test database
* Added ability to specify alternate config file at the command line
* New method to find ContentResource by UUID
* Metaobj and Wiki support
* More graceful fails when Users, Sites, and AuthzRealms are not found
* Fixed Sequel 4.0 sql_or -> or support

### 0.5.2 ###

*Released 2012-10-11*

* Added JSON output option for lookups
* Proof-of-concept sin shell
* Alias support

### 0.5.1 ###

*Released 2012-10-07*

* Added new private-message lookup type
* Adjusted to new use of the 'author' field in PrivateMessage/ForumPost table
* Now including post body with forum-post
* Added new pending-quiz-access-control and published-quiz-access-control types
* Integrated access control objects with quiz object output
* New deleted-content-resource type
* New rake task to load test fixtures into database
* Added to_csv method with field selection to SakaiObject
* Added more use cases to POC query mode
* Added field selection to lookup via --fields=x,y,z

### 0.5.0 ###

*Released 2012-06-21*

* Added --full-children and --all options for content
* Refactored rake tasks into one directory
* Added a barebones proof-of-concept-only query mode
* Removed quiz, quiz-section, and quiz-item types due to ID overlap risk
* Added new special modes to intercept calls for these removed object types
* Added published-quiz and pending-quiz types
* Added published-quiz-section and pending-quiz-section types
* Added published-quiz-item and pending-quiz-item types
* Added --texts record listing to published-quiz-item and pending-quiz-item
* Added --items record listing to quiz
* Better summaries for items in quiz-sections and quizzes
* Added a number of command line object type shortcuts
* Added random test fixture generation for user objects

### 0.4.6 ###

*Released 2012-05-19*

* Reinstated stub PrivateMessage class
* Removed defunct message.rb
* Re-added `count_by_date` method to PrivateMessage and ForumPost classes

### 0.4.5 ###

*Released 2012-05-14*

* CLI support for authz roles, realms, and functions
* Bugfix: fix syntax error using ContentCollection.find! on missing collections
* Bugfix: gracefully handle missing AnnouncementChannel for Site serializations

### 0.4.4 ###

*Released 2012-05-12*

* Standardized gradebook and gradebook-item APIs
* Renamed GradableObject class to GradebookItem
* CLI access to gradebook and gradebook-item objects

### 0.4.3 ###

*Released 2012-05-10*

* Tweaked site summary serialization to display creator EID
* Added attempted_at field to quiz attempt item serializations
* Added CLI support for announcement and announcement-channel
* Added clear_cache class method to all objects using caches
* Added Cache.clear_all class and method to SakaiInfo module

### 0.4.2 ###

*Released 2012-05-04*

* Fixed issue in assignment-submission for formatting entity dates
* Added status property to quizzes along with interpretation of the value
* Added support for provided users to User object (submitted by Will Humphries)
* Allowed .sakai-info file to point to another location for configuration

### 0.4.1 ###

*Released 2012-04-22*

* Improved MySQL support (from not working at all to mostly working)
* Removed properties performance hack which was inapplicable to MySQL
* Cleaned up site default serialization

### 0.4.0 ###

*Released 2012-04-21*

* CLI access to content
* Content binary entity parsing
* Command line tool name change to "sin"

## 0.3.5 ##

*Released 2012-04-02*

* Basic Forum, ForumThread, and ForumPost support in CLI
* Refactored Forum support in library
* Fixed join_role serialization for Site

## 0.3.4 ##

*Released 2012-03-09*

* Added --dbrow and --mod support to User
* User property lookup performance improvements

## 0.3.3 ##

*Released 2012-03-09*

* Fixed performance bug connecting to the database on every query
* Performance improvements (with possible downsides) in property lookups
* Group supports dbrow, improved serialization
* CLI now supports group objects

## 0.3.2 ##

*Released 2012-03-08*

* Site supports dbrow and ModProps, added published? method
* Page supports dbrow, improved serialization methods
* Tool supports dbrow, improved serialization methods
* CLI now supports page and tool objects

## 0.3.1 ##

*Released 2012-02-29*

* New objects: quiz-attempt, quiz-attempt-item, quiz-attempt-item-attachment
* More new objects: assignment, assignment-submission
* SakaiXMLEntity base class supports ModProps
* New standard CL arguments: --dbrow-only and --mod-details
* First steps towards a sqlite test infrastructure: schema creation
* First steps towards better Sakai DB docs: graphviz file for quiz tables

## 0.3.0 ##

*Released 2012-02-26*

* Support for new object types: question-pool, quiz, quiz-section, quiz-item
* New CLI options for logging and trace-debugging
* New object abstraction patterns allow for --dbrow and --mod options

## 0.2.1 ##

*Released 2012-02-25*

* Bugfix: pending quiz count was reporting incorrect numbers
* Bugfix: published quiz list was actually listing pending quizzes

## 0.2.0 ##

*Released 2012-02-24*


rakelib/data.rb
# data.rb
#   task definitions for generating test data
#
# Created 2012-05-21 daveadams@gmail.com
# Last updated 2012-10-12 daveadams@gmail.com
#
# https://github.com/daveadams/sakai-info
#
# This software is public domain. See LICENSE.
#

require 'net/http'
require 'base64'
require 'marky_markov'

DATADIR = File.join(TMPDIR, 'data')
FIRSTNAMESFILE = File.join(DATADIR, 'first_names.yml')
LASTNAMESFILE = File.join(DATADIR, 'last_names.yml')
RAWTEXTFILE = File.join(DATADIR, 'raw.txt')

FIXTUREDIR = File.join(TMPDIR, 'fixtures')
SAKAIUSERIDMAPFILE = File.join(FIXTUREDIR, "sakai_user_id_map.yml")

def create_data_dir
  print "Creating directory for raw data... "; STDOUT.flush
  system "mkdir -p #{DATADIR}"
  puts "OK"
end

def clear_data_dir
  print "Clearing data directory... "; STDOUT.flush
  system "rm -rf #{DATADIR}"
  puts "OK"
end

def create_fixture_dir
  print "Creating directory for fixtures... "; STDOUT.flush
  system "mkdir -p #{FIXTUREDIR}"
  puts "OK"
end

def clear_fixture_dir
  print "Clearing fixture directory... "; STDOUT.flush
  system "rm -rf #{FIXTUREDIR}"
  puts "OK"
end

def fetch_first_names
  if File.exist? FIRSTNAMESFILE
    return
  end
  if not File.exist? DATADIR
    create_data_dir
  end

  print "Fetching first names... "; STDOUT.flush
  first_names = []
  ["http://www.census.gov/genealogy/www/data/1990surnames/dist.female.first",
   "http://www.census.gov/genealogy/www/data/1990surnames/dist.male.first"].each do |url|
    content = Net::HTTP.get(URI(url))
    content.each_line do |line|
      first_names << line.split(/ /)[0].downcase.capitalize
    end
  end
  File.open(FIRSTNAMESFILE, "w") do |f|
    f.write(first_names.shuffle.to_yaml)
  end
  puts "OK"
end

def fetch_last_names
  if File.exist? LASTNAMESFILE
    return
  end
  if not File.exist? DATADIR
    create_data_dir
  end

  print "Fetching last names... "; STDOUT.flush
  last_names = []
  ["http://www.census.gov/genealogy/www/data/1990surnames/dist.all.last"].each do |url|
    content = Net::HTTP.get(URI(url))
    content.each_line do |line|
      last_names << line.split(/ /)[0].downcase.capitalize
    end
  end
  File.open(LASTNAMESFILE, "w") do |f|
    f.write(last_names.shuffle.to_yaml)
  end
  puts "OK"
end

def fetch_raw_text
  if File.exist? RAWTEXTFILE
    return
  end
  if not File.exist? DATADIR
    create_data_dir
  end

  content = ""
  print "Fetching raw text... "; STDOUT.flush
  [
   "http://www.gutenberg.org/cache/epub/1342/pg1342.txt",
   "http://www.gutenberg.org/cache/epub/11/pg11.txt",
   "http://www.gutenberg.org/cache/epub/74/pg74.txt",
   "http://www.gutenberg.org/cache/epub/98/pg98.txt",
  ].each do |url|
    lines = Net::HTTP.get(URI(url)).split("\n")
    content += lines.slice(150,(lines.length-600)).join("\n")
  end
  File.open(RAWTEXTFILE, "w") do |f|
    f.write(content)
  end
  puts "OK"
end

$markov_object = nil
def markov
  if $markov_object.nil?
    if not File.exist? RAWTEXTFILE
      fetch_raw_text
    end

    $markov_object = MarkyMarkov::TemporaryDictionary.new
    $markov_object.parse_file RAWTEXTFILE
  end
  return $markov_object
end

def generate_uuid
  # TODO: replace UUID generation with a ruby solution
  `uuidgen`.chomp
end

PWCHARS =
  ('a'..'z').to_a +
  ('A'..'Z').to_a +
  ('0'..'9').to_a +
  ['~','!','@','#','$','%','^','&','*','-','_','=','+',"'",'"',
   ';',':','.','>',',','<','/','?',"(",")","[","]","{","}",' ']

def generate_password
  (0..(8+rand(9))).map { PWCHARS[rand(PWCHARS.length)] }.join
end

def generate_random_letters
  letters = ('a'..'z').to_a
  (0..(3+rand(9))).map { letters[rand(letters.length)] }.join
end

def generate_random_words(count)
  markov.generate_n_words(count)
end

def generate_random_sentences(count)
  markov.generate_n_sentences(count)
end

def generate_random_paragraphs(count)
  paragraphs = []
  count.times do
    paragraphs << markov.generate_n_sentences(3+rand(5))
  end
  paragraphs.join("\n")
end

def generate_property_name
  case rand(3)
  when 0
    "prop-#{rand(10000)}-#{rand(100000)}"
  when 1
    "setting-#{rand(1000000)}"
  when 2
    generate_random_letters
  end
end

def generate_property_value
  case rand(3)
  when 0
    generate_uuid
  when 1
    generate_password
  when 2
    generate_random_words(3 + rand(3))
  end
end

TOOLS = {
  "osp.glossary" => "Glossary",
  "osp.matrix" => "Matrices",
  "osp.presentation" => "Portfolios",
  "sakai.announcements" => "Announcements",
  "sakai.assignment.grades" => "Assignments",
  "sakai.chat" => "Chat Room",
  "sakai.dropbox" => "Drop Box",
  "sakai.forums" => "Forums",
  "sakai.gradebook.tool" => "Gradebook",
  "sakai.iframe" => "Web Content",
  "sakai.messages" => "Messages",
  "sakai.news" => "News",
  "sakai.podcasts" => "Podcasts",
  "sakai.poll" => "Polls",
  "sakai.postem" => "Post'Em",
  "sakai.profile2" => "Profile",
  "sakai.resources" => "Resources",
  "sakai.rwiki" => "Wiki",
  "sakai.schedule" => "Schedule",
  "sakai.search" => "Search",
  "sakai.sections" => "Section Info",
  "sakai.site.roster" => "Roster",
  "sakai.siteinfo" => "Site Info",
  "sakai.sitestats" => "Site Stats",
  "sakai.syllabus" => "Syllabus",
}

def generate_page_tool_info
  registration = TOOLS.keys[rand(TOOLS.keys.length)]
  page_title = TOOLS[registration]
  tool_title = page_title

  if rand(500) == 499
    page_title = generate_random_words(1+rand(2))
  end

  if rand(5000) == 4999
    tool_title = generate_random_words(1+rand(2))
  end

  popup = 0
  if rand(1300) == 1000
    popup = 1
  end

  layout_hints = nil
  if rand(100) > 55
    layout_hints = "0,0"
  end

  {
    :page_id => generate_uuid,
    :tool_id => generate_uuid,
    :registration => registration,
    :page_title => page_title,
    :tool_title => tool_title,
    :popup => popup,
    :layout_hints => layout_hints,
  }
end

$all_user_ids = nil
def all_user_ids
  if $all_user_ids.nil?
    if not File.exist? SAKAIUSERIDMAPFILE
      STDERR.puts "#{SAKAIUSERIDMAPFILE} does not exist, please regenerate data."
      exit 1
    end
    $all_user_ids = YAML::load_file(SAKAIUSERIDMAPFILE).collect { |v| v[:user_id] }
  end
  $all_user_ids
end

def random_user_id
  all_user_ids[rand(all_user_ids.length)]
end

def generate_users(count)
  sakai_user_id_map =
    [{
       :user_id => "admin",
       :eid => "admin"
     }]
  sakai_user =
    [{
       :user_id => "admin",
       :email => "sakaiadmin@sample.com",
       :first_name => "Sakai",
       :last_name => "Administrator",
       :type => "admin",
       :pw => "VlAvQCY9NC86OSs=",
       :createdby => "admin",
       :createdon => (Time.now - (8*60*60*24*365)),
       :modifiedby => "admin",
       :modifiedon => (Time.now - rand(8*60*60*24*365)),
       :email_lc => "sakaiadmin@sample.com",
     }]
  sakai_user_property = []

  first_names = YAML.load_file(FIRSTNAMESFILE)
  last_names = YAML.load_file(LASTNAMESFILE)

  print "Generating fixtures for #{count} user records... "; STDOUT.flush
  count.times do
    user_id = generate_uuid
    first_name = first_names[rand(first_names.length)]
    last_name = last_names[rand(last_names.length)]
    eid = (case rand(7)
           when 0 then (first_name[0] + last_name)
           when 1 then (first_name + last_name[0])
           when 2 then (first_name[0] + last_name + (rand(100).to_s))
           when 3 then (first_name + last_name[0] + (rand(100).to_s))
           when 4 then (first_name + (rand(1000).to_s))
           when 5 then (last_name + (rand(1000).to_s))
           when 6 then (first_name[0] + last_name[0] + (rand(10000).to_s))
           end).downcase
    email = eid + "@sample.com"
    type = if rand(100) > 80
             "guest"
           else
             "registered"
           end
    createdby = "admin"
    modifiedby = "admin"
    ago = rand(8*60*60*24*365)
    createdon = (Time.now - ago)
    modifiedon = (Time.now - rand(ago))
    if type == "guest"
      eid = email
      createdby = sakai_user_id_map[rand(sakai_user_id_map.length)][:user_id]
      modifiedby = sakai_user_id_map[rand(sakai_user_id_map.length)][:user_id]
    end
    pw = Base64.encode64(generate_password).chomp

    sakai_user_id_map << {
      :user_id => user_id,
      :eid => eid
    }

    sakai_user << {
      :user_id => user_id,
      :email => email,
      :first_name => first_name,
      :last_name => last_name,
      :type => type,
      :pw => pw,
      :createdby => createdby,
      :createdon => createdon,
      :modifiedby => modifiedby,
      :modifiedon => modifiedon,
      :email_lc => email
    }

    rand(3).times do
      sakai_user_property << {
        :user_id => user_id,
        :name => generate_property_name,
        :value => generate_property_value,
      }
    end
  end

  File.open(File.join(FIXTUREDIR, "sakai_user_id_map.yml"), "w") do |f|
    f.write(sakai_user_id_map.to_yaml)
  end

  File.open(File.join(FIXTUREDIR, "sakai_user.yml"), "w") do |f|
    f.write(sakai_user.to_yaml)
  end

  File.open(File.join(FIXTUREDIR, "sakai_user_property.yml"), "w") do |f|
    f.write(sakai_user_property.to_yaml)
  end
  puts "OK"
end

def generate_sites(count)
  sakai_site = []
  sakai_site_property = []
  sakai_site_page = []
  sakai_site_page_property = []
  sakai_site_tool = []
  sakai_site_tool_property = []

  # TODO: generate default sites
  # TODO: generate user workspaces

  print "Generating fixtures for #{count} site records... "; STDOUT.flush
  count.times do
    site_id = generate_uuid
    site_title = generate_random_words(3)
    short_desc = generate_random_sentences(2)
    long_desc = generate_random_paragraphs(1+rand(3))
    published = 1
    if rand(100) > 90
      published = 0
    end

    joinable = 0
    if rand(100) > 95
      joinable = 1
    end

    pubview = 0
    if rand(100) > 80
      pubview = 1
    end

    site_type = "course"
    if rand(100) > 80
      site_type = "project"
      if rand(100) > 95
        site_type = "portfolio"
      end
    end

    join_role = nil
    if joinable == 1
      join_role = "Guest"
      if rand(100) > 50
        join_role = "Participant"
        if site_type == "course"
          join_role = "Student"
        end
      end
    end

    custom_page_ordered = 1
    if rand(100) > 60
      custom_page_ordered = 0
    end

    ago = rand(8*60*60*24*365)
    createdon = (Time.now - ago)
    modifiedon = createdon
    if rand(100) > 40
      modifiedon = (Time.now - rand(ago))
    end

    createdby = random_user_id
    modifiedby = createdby
    if modifiedon != createdon
      if rand(100) > 50
        modifiedby = random_user_id
      end
    end

    sakai_site << {
      :site_id => site_id,
      :title => site_title,
      :type => site_type,
      :short_desc => short_desc,
      :description => long_desc,
      :icon_url => nil,
      :info_url => nil,
      :skin => nil,
      :published => published,
      :joinable => joinable,
      :pubview => pubview,
      :join_role => join_role,
      :createdby => createdby,
      :modifiedby => modifiedby,
      :createdon => createdon,
      :modifiedon => modifiedon,
      :is_special => 0,
      :is_user => 0,
      :custom_page_ordered => custom_page_ordered,
    }

    rand(3).times do
      sakai_site_property << {
        :site_id => site_id,
        :name => generate_property_name,
        :value => generate_property_value,
      }
    end

    site_order = 1
    site_tools = []
    (6+rand(5)).times do
      site_order += 1
      info = nil
      # ensure we don't duplicate any tools
      # TODO: exclude tools that can be duplicated from this rule
      while info.nil?
        info = generate_page_tool_info
        if site_tools.include?(info[:registration])
          info = nil
        end
      end
      site_tools << info[:registration]

      sakai_site_page << {
        :page_id => info[:page_id],
        :site_id => site_id,
        :title => info[:page_title],
        :layout => 0,
        :site_order => site_order,
        :popup => info[:popup],
      }

      rand(3).times do
        sakai_site_page_property << {
          :site_id => site_id,
          :page_id => info[:page_id],
          :name => generate_property_name,
          :value => generate_property_value,
        }
      end

      sakai_site_tool << {
        :tool_id => info[:tool_id],
        :page_id => info[:page_id],
        :site_id => site_id,
        :registration => info[:registration],
        :page_order => 1,
        :title => info[:tool_title],
        :layout_hints => info[:layout_hints],
      }

      rand(3).times do
        sakai_site_tool_property << {
          :site_id => site_id,
          :tool_id => info[:tool_id],
          :name => generate_property_name,
          :value => generate_property_value,
        }
      end
    end
  end

  File.open(File.join(FIXTUREDIR, "sakai_site.yml"), "w") do |f|
    f.write(sakai_site.to_yaml)
  end

  File.open(File.join(FIXTUREDIR, "sakai_site_property.yml"), "w") do |f|
    f.write(sakai_site_property.to_yaml)
  end

  File.open(File.join(FIXTUREDIR, "sakai_site_page.yml"), "w") do |f|
    f.write(sakai_site_page.to_yaml)
  end

  File.open(File.join(FIXTUREDIR, "sakai_site_page_property.yml"), "w") do |f|
    f.write(sakai_site_page_property.to_yaml)
  end

  File.open(File.join(FIXTUREDIR, "sakai_site_tool.yml"), "w") do |f|
    f.write(sakai_site_tool.to_yaml)
  end

  File.open(File.join(FIXTUREDIR, "sakai_site_tool_property.yml"), "w") do |f|
    f.write(sakai_site_tool_property.to_yaml)
  end
  puts "OK"
end

namespace :data do
  desc "Pull raw data to build test database"
  task :pull do
    fetch_first_names
    fetch_last_names
    fetch_raw_text
  end

  desc "Generate random fixture data for testing"
  task :generate => [:pull] do
    if not File.exist? FIXTUREDIR
      create_fixture_dir
    end

    if (not (File.exist?(File.join(FIXTUREDIR, "sakai_user_id_map.yml")) and
             File.exist?(File.join(FIXTUREDIR, "sakai_user.yml")) and
             File.exist?(File.join(FIXTUREDIR, "sakai_user_property.yml"))))
      generate_users(50)
    end

    if (not (File.exist?(File.join(FIXTUREDIR, "sakai_site.yml")) and
             File.exist?(File.join(FIXTUREDIR, "sakai_site_property.yml")) and
             File.exist?(File.join(FIXTUREDIR, "sakai_site_page.yml")) and
             File.exist?(File.join(FIXTUREDIR, "sakai_site_page_property.yml")) and
             File.exist?(File.join(FIXTUREDIR, "sakai_site_tool.yml")) and
             File.exist?(File.join(FIXTUREDIR, "sakai_site_tool_property.yml"))))
      generate_sites(10)
    end
  end

  desc "Clear raw data and fixture data"
  task :clear do
    clear_fixture_dir
    clear_data_dir
  end
end

